cumulus/pallets/aura-ext/src/consensus_hook.rs

```diff
@@ -54,8 +53,23 @@ where
 		let velocity = V.max(1);
 		let relay_chain_slot = state_proof.read_slot().expect("failed to read relay chain slot");
 
-		let (slot, authored) =
-			pallet::SlotInfo::<T>::get().expect("slot info is inserted on block initialization");
+		let (relay_chain_slot, authored_in_relay) = match pallet::RelaySlotInfo::<T>::get() {
+			Some((slot, authored)) if slot == relay_chain_slot => (slot, authored + 1),
+			Some((slot, _)) if slot < relay_chain_slot => (relay_chain_slot, 1),
+			Some(..) => {
+				panic!("slot moved backwards")
+			},
+			None => (relay_chain_slot, 1),
+		};
+
+		// We need to allow one additional block to be built to fill the unincluded segment.
+		if authored_in_relay > velocity + 1 {
+			panic!("authored blocks limit is reached for the slot: relay_chain_slot={relay_chain_slot:?}, authored={authored_in_relay:?}, velocity={velocity:?}");
+		}
+
+		pallet::RelaySlotInfo::<T>::put((relay_chain_slot, authored_in_relay));
```

Comment:
```suggestion
		pallet::RelaySlotInfo::<T>::put((relay_chain_slot, authored_in_relay + 1));
```

If we do this and also have above return `0` as `authored_in_relay` for a new slot, we don't need the `velocity + 1`?